### PR TITLE
#14. Message for records which does not exist in DB-IDs 203,204,29 do…

### PR DESCRIPTION
**Pull request by BhagyashreeBhoyar**
…esn’t exist in database and not processed (UI).

### CODE REVIEW

File: src/components/UploadOptInFile.jsx [hunk 1]
```diff
@@ -8,6 +8,7 @@ import {
   isOptInSuccess,
   getFailOptIn,
   isUploadOptInFailed,
+  idNotPresent,
```
Comment (pepcus-pratik) [inline src/components/UploadOptInFile.jsx]: rename this to `isIdUnavailable`

File: src/components/UploadOptInFile.jsx [hunk 2]
```diff
@@ -8,6 +8,7 @@ import {
   isOptInSuccess,
```
Comment (pepcus-pratik) [inline src/components/UploadOptInFile.jsx]: Update the proptypes too

File: src/reducers/studentRegistrationReducer.js [hunk 3]
```diff
@@ -466,4 +475,8 @@ export const isMarkAttendanceFailed = state => state.allStudentsDataReducer.isMa
 
 export const isMarkOptInOrOptOutFailed = state => state.allStudentsDataReducer.isMarkOptInOrOptOutFailed;
 
-export const isUpdateIdCardStatusFailed = state => state.allStudentsDataReducer.isUpdateIdCardStatusFailed;
\ No newline at end of file
+export const isUpdateIdCardStatusFailed = state => state.allStudentsDataReducer.isUpdateIdCardStatusFailed;
+
+export const idNotExist = state => state.allStudentsDataReducer.idNotExist;
+
+export const idNotPresent = state => state.allStudentsDataReducer.idNotPresent;
```
Comment (pepcus-pratik) [inline src/reducers/studentRegistrationReducer.js]: Add JSDoc comments

File: src/components/UploadOptInFile.jsx (hunk 1 above)
Comment (BhagyashreeBhoyar) [reply src/components/UploadOptInFile.jsx]: Done.

File: src/reducers/studentRegistrationReducer.js (hunk 3 above)
Comment (BhagyashreeBhoyar) [reply src/reducers/studentRegistrationReducer.js]: Done.

File: src/components/UploadOptInFile.jsx (hunk 2 above)
Comment (BhagyashreeBhoyar) [reply src/components/UploadOptInFile.jsx]: Done.

File: src/components/UploadOptInFile.jsx [hunk 4]
```diff
@@ -196,16 +199,36 @@ class UploadOptInFile extends Component {
 
   }
 }
+
+UploadOptInFile.propTypes = {
+  resetIsOptInSuccessAction: PropTypes.func,
```
Comment (pepcus-pratik) [inline src/components/UploadOptInFile.jsx]: `actions` can be marked as `isRequired` in propTypes

**Pull request merged**